packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1591,6 +1591,53 @@ abstract class _RenderPhysicalModelBase<T> extends _RenderCustomClip<T> {
     description.add(DiagnosticsProperty<Color>('color', color));
     description.add(DiagnosticsProperty<Color>('shadowColor', color));
   }
+
+  @protected
+  Path get clipPath;
+
+  @override
+  @mustCallSuper
+  void paint(PaintingContext context, Offset offset) {
+    assert(() {
+      RenderObject ancestor = parent;
+      while (ancestor != null) {
+        if (ancestor is _RenderPhysicalModelBase) {
+          // We're the direct child of another physical model. That is the one
```

Comment:
Can there not be conflict within a PhysicalModelShape:

```
--------------------- Material
          |                       
          |   ------------------ Material
          |           |
----------------------------------- Material
                  |
----------------------------------- Material
```

Reply:
I'm not sure I'm understanding your diagram completely, but if a `Material` is a child of another `Material`, we only need to check the parent material since if the child overlaps with something else, the parent does too.